migreringer/src/main/resources/db/migration/defaultDS/4.1/V4.1_24__TFP-5383-aktivitetskrav_arbeid.sql

```diff
@@ -0,0 +1,117 @@
+create sequence SEQ_GR_AKTIVITETSKRAV_ARBEID
+    increment by 50
+    nocache;
+
+create sequence SEQ_AKTIVITETSKRAV_ARBEID_PERIODER
+    increment by 50
+    nocache;
+
+create sequence SEQ_AKTIVITETSKRAV_ARBEID_PERIODE
+    increment by 50
+    nocache;
+
+create table AKTIVITETSKRAV_ARBEID_PERIODER
+(
+    OPPRETTET_TID TIMESTAMP(3) default systimestamp not null,
+    OPPRETTET_AV VARCHAR2(20 char) default 'VL' not null,
+    ID NUMBER(19) not null
+        constraint PK_AKTIVITETSKRAV_ARBEID_PERIODER
+        primary key
+)
+    /
+
+comment on table AKTIVITETSKRAV_ARBEID_PERIODER is 'Aggregering av informasjon om aktivitetskrav arbeid'
+/
+
+comment on column AKTIVITETSKRAV_ARBEID_PERIODER.ID is 'Primærnøkkel'
+/
+
+create table GR_AKTIVITETSKRAV_ARBEID
+(
+    AKTIV VARCHAR2(1 char) default 'J' not null
+        constraint GR_AKTIVITETSKRAV_ARBEID
+        check (aktiv IN ('J', 'N')),
+    ENDRET_TID TIMESTAMP(3),
+    ENDRET_AV VARCHAR2(20 char),
+    OPPRETTET_TID TIMESTAMP(3) default systimestamp not null,
+    OPPRETTET_AV VARCHAR2(20 char) default 'VL' not null,
+    AKTIVITETSKRAV_ARBEID_PERIODER_ID NUMBER(19)
+        constraint FK_GR_AKTIVITETSKRAV_ARBEID_2
+        references AKTIVITETSKRAV_ARBEID_PERIODER,
+    BEHANDLING_ID NUMBER(19) not null
+        constraint FK_GR_AKTIVITETSKRAV_ARBEID_1
+        references BEHANDLING,
+    ID NUMBER(19) not null
+        constraint PK_GR_AKTIVITETSKRAV_ARBEID
+        primary key
+)
+    /
+
+comment on table GR_AKTIVITETSKRAV_ARBEID is 'Behandlingsgrunnlag for aktivitetskrav arbeid'
+/
+
+comment on column GR_AKTIVITETSKRAV_ARBEID.AKTIV is 'Om valg er aktivt'
+/
+
+comment on column GR_AKTIVITETSKRAV_ARBEID.AKTIVITETSKRAV_ARBEID_PERIODER_ID is 'Foreign key til AKTIVITETSKRAV_ARBEID_PERIODER'
+/
+
+comment on column GR_AKTIVITETSKRAV_ARBEID.BEHANDLING_ID is 'Foreign key til behandling'
+/
+
+comment on column GR_AKTIVITETSKRAV_ARBEID.ID is 'PK'
+/
+
+create index IDX_GR_AKTIVITETSKRAV_ARBEID_1
+    on GR_AKTIVITETSKRAV_ARBEID (BEHANDLING_ID)
+    /
+
+create index IDX_GR_AKTIVITETSKRAV_ARBEID_2
+    on GR_AKTIVITETSKRAV_ARBEID (AKTIVITETSKRAV_ARBEID_PERIODER_ID)
+    /
+
+create table AKTIVITETSKRAV_ARBEID_PERIODE
+(
+    OPPRETTET_TID TIMESTAMP(3) default systimestamp not null,
+    OPPRETTET_AV VARCHAR2(20 char) default 'VL' not null,
+    AKTIVITETSKRAV_ARBEID_PERIODER_ID NUMBER(19) not null
+        constraint FK_AKTIVITETSKRAV_ARBEID_PERIODE_1
+        references AKTIVITETSKRAV_ARBEID_PERIODER,
+    TOM DATE not null,
+    FOM DATE not null,
+    ORG_NR VARCHAR2(100 char) not null,
+    SUM_STILLINGSPROSENT NUMBER(5) not null,
+    SUM_PERMISJONSPROSENT NUMBER(5) not null,
```

Comment:
Du klarer deg med NUMBER(3) siden du ikke vil ha med desimaler - vanligvis bruker man prosent number(5,2)

Reply:
Når det er snakk om stillingsprosent eller permisjonsprosent bør vi ikke da ha med desimaler?

Reply:
Jeg valgte å ha desimaler

Reply:
Da kan du bruke @Embeddable Stillingsprosent med AttributeOverride. Den sørger for normalisering til 2 desimaler

Det har vært brukt desimaler i stillingsprosent - men har det noen reell nytte her?

Reply:
Det fins tilfeller hvor stillingsprosent har desimaler i aa-reg - unntakstilfeller selvfølgelig. Det er 5 tilfeller av 748 (logg-tilfellene). Det har antagelig ingen betydning når vi summerer uansett. Eneste er at det kan ha nytte når vi viser saksbehandler stillingsprosent per orgnr, og de får tallene til å stemme med det de ser i registeret?